### DISCUSSION TITLE
Customize the summary of metrics in wandb table

### DISCUSSION THREAD

**Discussion by GillianGrayson** [ashleve/lightning-hydra-template#170] [Q&A] [answered]
Hi!

Is there an easy way to customize the summary of metrics in wandb table?

By default, the summary is set to the last value you logged for each key.
But the last value is not always optimal, because there is a non-zero `patience` parameter in the `EarlyStopping` callback.
In the result table are the last (not optimal values).
How to deal with it in the simplest way?

Thanks!

P.S.: The best template repository for PyTorch!

**Comment by GillianGrayson** [comment, marked as answer]
One possible [way](https://gitbook-docs.wandb.ai/guides/track/log#summary-metrics) :

```python
import wandb

...

    def on_fit_start(self) -> None:
        wandb.define_metric(f"train/loss", summary='min')
        wandb.define_metric(f"val/loss", summary='min')
        wandb.define_metric(f"test/loss", summary='min')
        wandb.define_metric(f"train/acc", summary='max')
        wandb.define_metric(f"val/acc", summary='max')
        wandb.define_metric(f"test/acc", summary='max')
```

**Reply by ashleve** [reply]
Thanks! Didn't know about that
